releng/org.eclipse.n4js.releng.utils/src/org/eclipse/n4js/antlr/n4js/AutomaticSemicolonInjector.java

```diff
@@ -62,7 +62,7 @@ private String addCustomMethodsToParser(String grammarContent) {
 		return replace(grammarContent, "@members {",
 				"@members {\n\n" +
 						"// injected by AutomaticSemicolonInjector\n" +
-						"protected boolean forcedRewind(int marker) { return true; } // overridden in subtype\n" +
+						"protected boolean forcedRewind(int marker, boolean advance) { return true; } // overridden in subtype\n" +
```

Comment:
😨

Reply:
It was a shortcut. I'll make this two methods :)

Reply:
Actually my comment was related to the idea that source code is modified with string replace. I did not see that before in our code base and was mildly shocked.

Reply:
It's a beauty, I know ;)

Nowadays we could customize the grammar templates instead. But would still mean code generation == code in strings.